index.js

```diff
@@ -1,14 +1,33 @@
 'use strict'
 
 const fp = require('fastify-plugin')
-const { performance } = require('perf_hooks')
+const { performance, PerformanceObserver } = require('perf_hooks')
 const summary = require('summary')
 
 const ONSEND = 'on-send-'
 const PREHANDLER = 'pre-handler-'
 const ROUTES = 'fastify-routes:'
 
 module.exports = fp(async function (fastify, opts) {
+  let observedEntries = {}
+  const obs = new PerformanceObserver((items) => {
+    const fetchedItems = items.getEntries()
+    for (let i = 0; i < fetchedItems.length; i++) {
+      const e = fetchedItems[i]
+      if (e.name.indexOf(ROUTES) === 0) {
+        const key = e.name.split(':')[1]
+        if (observedEntries[key]) {
+          observedEntries[key].push(e.duration)
+        } else {
+          observedEntries[key] = [e.duration]
+        }
+      }
+    }
+
+    performance.clearMarks()
+  })
+  obs.observe({ entryTypes: ['measure'] })
+
   fastify.addHook('preHandler', function (request, reply, next) {
```

Comment:
while we are updating this, we might want to move this to `onRequest`. It is more accurate.